presto-main/src/main/java/io/prestosql/execution/CommentTask.java

```diff
@@ -47,12 +50,24 @@ public String getName()
     }
 
     @Override
-    public ListenableFuture<?> execute(Comment statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
+    public ListenableFuture<?> execute(
+            Comment statement,
+            TransactionManager transactionManager,
+            Metadata metadata,
+            AccessControl accessControl,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
     {
         Session session = stateMachine.getSession();
 
         if (statement.getType() == Comment.Type.TABLE) {
             QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
+            Optional<QualifiedObjectName> redirectedTableName = metadata.redirectTable(session, tableName, true);
+            if (redirectedTableName.isPresent()) {
```

Comment:
Move the warning inside of the `redirectTable()` method and include the source and destination in the warning message. This keeps the code simpler and we don't need to add warning collector everywhere.

Then this code can become
```java
tableName = metadata.redirectTable(session, tableName, true).orElse(tableName);
```
Or perhaps add a variant that returns either the redirection or the original.